src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/ExtensionDataTests.cs

```diff
@@ -1239,7 +1239,7 @@ public static void CustomJsonElementConverterInExtensionProperty()
 
             ClassWithExtensionPropertyAsJsonElement obj = JsonSerializer.Deserialize<ClassWithExtensionPropertyAsJsonElement>(Json, options);
             JsonElement overflowProp = obj.MyOverflow["hello"];
-            Assert.Equal(JsonValueKind.String, overflowProp.ValueKind);
+            Assert.Equal(JsonValueKind.Undefined, overflowProp.ValueKind);
```

Comment:
What causes the change here? Do we need to document a breaking change?

Reply:
This test change is correct. There was a bug previously where the custom converter was only used on write, not read.